File: leetcode/problems/1.two-sum/js/approach3.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let temp = {};
    let results;
    
    for (let i = 0; i < nums.length; i++) {
        if (temp[target - nums[i]] !== undefined) {
            results = [temp[target - nums[i]], i]
        }
        
        temp[nums[i]] = i
    }
    
    return results
};

module.exports = twoSum

File: leetcode/problems/1.two-sum/go/approach3.go
package solution

func twoSum(nums []int, target int) []int {
	temp := make(map[int]int)

	for i, value := range nums {

		if j, ok := temp[target-value]; ok {
			return []int{j, i}
		}

		temp[value] = i
	}

	return nil
}
